### PR TITLE
Update: Add true curves APIs to `Add graphics with renderer`

### PR DESCRIPTION
**Pull request by yo1995**
This is the first U12 API into `v.next` branch. Changes

- Add true curve APIs according to `common-samples/issues/2819`
- Update the screenshot with a heart shape
- Update docs to reflect the changes
- Rename the src files to follow the naming convention
- Remove `redirect_from` as it is no longer valid

### CODE REVIEW

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:131]: Since you've updated the other variable names, I recommend `sr` be changed as well.

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md:13]:
```suggestion
Pan and zoom on the map to view graphics for points, lines, polygons, and shapes with curved segments which are stylized using renderers.
```
I think it's been a while since we've updated this sample and README. A slight change would keep this up to date with our current guidelines.

**Review by vquach2404 — COMMENTED** [review]
Since the graphics are in the same area, it may be good to set the viewpoint to focus on the graphics. Overall looking good!

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:82]: I think the `make___GraphicsOverlay` functions could be generalized so you can pass the symbol type and use a switch statement. What do you think?

Comment (yo1995) [reply arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:82]: Good point. I mostly copy pasted the code from the original implementation. Due to renaming of the file they don't show as diff but as deletions, so there is no way to compare them on GitHub.

I think this sample is intended to face those more beginner-level developers. So I intentionally didn't make big structural changes to the code, so that they can be copied and pasted easily. E.g. when someone want to test a single symbol, they can just copy the 2 methods (`makeSomeGraphicsOverlay` and `addGraphicsOverlays`) into a new file and see them renders.

I did extract them into `makeSomeGraphicsOverlay` methods though because otherwise the original `addGraphicsOverlay` method would be too long after adding the new curvature graphics. 😅 

I'll leave this discussion open to Phil and Sarat.

**Comment by yo1995** [comment]
Thanks for the review. I guess I'll hold off making changes to this PR until the design update has been merged - otherwise there are too many back and forth to keep them in sync 😅 .

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md:15]: I didn't notice this before since no changes were made to this section, but the list should be enumerated. I also wonder if some explanation should be added to go into depth about the added API. Seems a bit odd to have a long list of Relevant API but disregard half of them in this section.

Comment (yo1995) [reply arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md:15]: Do you mean to change it to ...?
> Pan and zoom on the map to view graphics for points, lines, **and** polygons (including polygons with curve segments) which are stylized using renderers.

> I also wonder if some explanation should be added to go into depth about the added API. Seems a bit odd to have a long list of Relevant API but disregard half of them in this section.

It would have been a good question to ask in the design update PR. 

IMO though it is OK to omit them from the `How it works` section. This sample is intended for exhibiting how to apply renderers to graphics overlays, not how to construct geometries. True curve APIs are only included in the code so that user can find example snippet when they search for these keywords. @TADraeseke may chime in here.

Comment (vquach2404) [reply arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md:15]: I think others may have missed this section in the design update as well since this section wasn't highlighted in either PRs. Feel free to request Phil's and Sarat's reviews so they can chime in as well

**Review by vquach2404 — DISMISSED** [review]

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md
Comment (TADraeseke) [reply arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md:15]: Oh yeah! Should be enumerated according to our [README template](https://devtopia.esri.com/runtime/common-samples/wiki/README-Template#how-it-works). 

As to Relevant API. Really good point @vquach2404. This sample's main purpose is to show use of a renderer and it's been co-opted to show a curve since it's an easy win until we build out more functionality next release.

Let's not worry about changing the steps too much now (though they should be enumerated). This sample will serve as a basic story that we support curves, but we'll go into detail on how in another sample next release

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift
Comment (philium) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:155]: `reduce` is really suited more toward value types. For something like this, it seems a bit overkill. You might consider doing something a bit simpler:

```swift
let heart = AGSMutablePart(spatialReference: spatialReference)
[leftCurve, leftArc, rightArc, rightCurve].forEach { heart.add($0) }
```

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift
Comment (philium) [inline arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:169]: If it were me, I would do away with `addGraphicsOverlay()` and instead have methods for creating each of the overlays and then call them in the `mapView` property observer:

```swift
mapView.graphicsOverlays.addObjects(from: [
    makePointGraphicsOverlay(),
    makeLineGraphicsOverlay(),
    makePolygonGraphicsOverlay(),
    makeCurveGraphicsOverlay()
])
```

File: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift
Comment (yo1995) [reply arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift:121]: `AGSPoint`'s initializer takes `Double` for coordinates, thus I used double for the side length, instead of the conventional `CGFloat` (for core graphics classes such as `CGRect` and `CGSize`) to avoid casting back to double.

**Review by philium — CHANGES_REQUESTED** [review]
Please update the description in `ContentPList.plist`.

**Review by philium — DISMISSED** [review]

**Comment by yo1995** [comment]
> Please update the description in `ContentPList.plist`.

Thanks for catching this! Though I wasn't sure why this sample was excluded from the description checker script in the first place 🤦‍♂️  https://github.com/Esri/arcgis-runtime-samples-ios/blob/ee1bbe9b20a009770c12a3f2fb048edf552115e9/Scripts/CI/README_Metadata_StyleCheck/description_differ.py#L33

I'll request Sarat's verification.

**Comment by philium** [comment]
> Though I wasn't sure why this sample was excluded from the description checker script in the first place 🤦‍♂️

Can that exception be removed? And if so, could that be done in this PR?

**Review by saratchandrakarumuri — APPROVED** [review]
Looks good

**Review by philium — APPROVED** [review]

**Comment by yo1995** [comment]
Thank you all!

**Pull request merged**
